src/main/java/io/javalin/core/compression/DynamicCompressionStrategy.kt

```diff
@@ -0,0 +1,87 @@
+package io.javalin.core.compression
+
+import org.meteogroup.jbrotli.libloader.BrotliLibraryLoader
+import io.javalin.Javalin.log
+
+/**
+ * This class is a settings container for Javalin's dynamic content compression.
+ *
+ * It is used by DynamicCompressionHandler to determine the encoding and parameters that should be used
+ * when encoding a dynamic response from the server.
+ *
+ * @see DynamicCompressionHandler
+ *
+ * @param brotliEnabled Default: false
+ * @param gzipEnabled   Default: true
+ * @param brotliLevel   Brotli compression level. Higher means better (but slower) compression.
+ *                      Range 0..11, Default: 4
+ * @param gzipLevel     Gzip compression level. Higher means better (but slower) compression.
+ *                      Range 0..9, Default: 6
+ */
+class DynamicCompressionStrategy @JvmOverloads constructor(
+        brotliEnabled: Boolean = false,
+        gzipEnabled: Boolean = true,
+        brotliLevel: Int = 4,
+        gzipLevel: Int = 6) {
```

Comment:
This constructor feels a bit too complicated to me. Would it be possible to either treat `0` as off (I don't know if that's the correct behavior)? Or create a simple data class that takes the level, then let that be `null` if it's disabled?

Reply:
Then you could also do your input validation in the class which takes the level.

Reply:
0 is a valid compression setting for both algorithms:
- Setting 0 on gzip will perform no compression, but will still set the content-encoding header to gzip.
- Setting 0 on brotli still performs significant compression

The data class would be the way to go in this case. I do agree the constructor is a bit clunky, now that I think about it. For example, if you wanted to change gzip's compression level, you are automatically forced to set brotli's as well. I don't like this.

Would something like "DynamicCompressionStrategy(Brotli(4), Gzip(6))" work? (without level params for defaults)

Reply:
Yes, that looks good!